src/rtop/rtop.ml

```diff
@@ -0,0 +1,31 @@
+(* Try to read the HOME environment
+ * if it is not set we assume it's windows and fallback to USERPROFILE *)
+let home = match Sys.getenv_opt "HOME" with
+| Some(p) -> p
+| None -> Sys.getenv "USERPROFILE"
+in
+
+let () = Unix.openfile (home ^ "/.utoprc") [] 0o640
+|> Unix.close in
+
+let () = Unix.openfile (home ^ "/.utop-history") [] 0o640
+|> Unix.close in
+
+(* Get the dir of the current executable *)
+let dir = Filename.dirname Sys.argv.(0) in
+
+(* If there is only 1 arg it's the exeuctable and we can remove that *)
+let argv = match Array.length Sys.argv with
+| 1 -> [||]
+| argLength -> Array.sub Sys.argv 1 argLength
+in
+
+let args = if Array.length argv == 1 && Sys.argv.(0) == "stdin" then
```

Comment:
Change `Sys.argv` to `argv`

Reply:
You also need to change the `==` in the string comparison to `=`, otherwise it's always false

```ocaml
utop # "foo" == "foo";;
- : bool = false
```